### DISCUSSION TITLE
packet queue is empty, aborting

### DISCUSSION THREAD

**Discussion by vladigor777** [miguelgrinberg/Flask-SocketIO#1714] [Q&A]
Hello! I often get this message from my client! Could you tell me please, how can I hide this message?

**Comment by vladigor777** [comment]
while True:
        client = socketio.AsyncClient()

        @client.event
        def handler_func(msg, order):
            return get_best_bid_ask_from_orderbook(msg, order, pair, orderbook_dict, rates_dict, hit_ready_amount, hit_ready_quantity,
                                                   hit_ready_volume)



        @client.event
        async def connect():
            pass

        @client.event
        def connect_error():
            print("The connection failed!")

        @client.event
        async def disconnect():
            global is_disconnected
            is_disconnected = True
            # await client.connect('https://socket.stex.com', transports=['websocket'])
            # await client.sleep(1)

        token = '555'

        await client.connect('https://socket.stex.com', transports=['websocket'])
        await client.sleep(1)
        global is_disconnected
        is_disconnected = False
        orderbook_dict = await get_stex_orderbook(pair_id)
        await client.emit('subscribe', {
                        'channel': f'sell_data{pair_id}',
                        'auth': {'headers': {'Authorization': 'Bearer ' + token}}
                    })
        await client.emit('subscribe', {
            'channel': f'buy_data{pair_id}',
            'auth': {'headers': {'Authorization': 'Bearer ' + token}}
        })
        while True:
            client.on("App\Events\GlassRowChanged", handler_func)
            await client.sleep(0.1)
            if is_disconnected:
                break

**Reply by miguelgrinberg** [reply]
This often means that the server isn't sending the correct packets according to the Socket.IO specification. What server is this?

**Reply by vladigor777** [reply]
Cryptoexchange

**Reply by vladigor777** [reply]
Is it possible somehow not to receive these messages? Server is exchange https://app.stex.com/ru/login

**Reply by vladigor777** [reply]
@miguelgrinberg

**Reply by miguelgrinberg** [reply]
You may want to try using the regular (non-async) client to rule any blocking problems on your part in the client. Beyond that I don't have any other suggestions, I'm not familiar with this server.
